### PR TITLE
Summary tab - Ability to add contact to case

### PR DESCRIPTION
**Pull request by acedeywin**
Primary reviewer: @stephenhand 

## Description

- New feature: Modifying the existing “Add to New Case” button on the Summary tab of an Active Contact.

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Related Issues
Fixes CHI-1972

### Verification steps

- Create an offline contact 
-  Verify that we have a new "Add to New Case” drop-up button with "New Case" and "Existing Case" as options 
- The "New Case” button should behave the same as the previous “Add Contact to Case” button
- The “Existing Case” button should take you to the Search tab

### CODE REVIEW

**Review by stephenhand — COMMENTED** [review]
You should merge up from master because the other code that adds a caseId property so that conflict can be resolved, because I think that's causing the E2E test fails

File: plugin-hrm-form/src/types/types.ts [hunk 1]
```diff
@@ -168,6 +168,7 @@ export type Contact = {
   accountSid: string;
   twilioWorkerId: string;
   number: string;
+  caseId?: number | null;
```
Comment (stephenhand) [inline plugin-hrm-form/src/types/types.ts:171]:
```suggestion
  caseId?: number;
```

You only need the ? for optional parameters

File: plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx [hunk 2]
```diff
@@ -140,33 +168,68 @@ const BottomBar: React.FC<
...
-                >
-                  <FolderIcon style={{ fontSize: '16px', marginRight: '10px' }} />
-                  <Template code="BottomBar-AddContactToNewCase" />
-                </StyledNextStepButton>
+            <AddNewCaseDropdown
+              handleNewCaseType={handleSubmitIfValid(handleOpenNewCase, onError)}
+              handleExistingCaseType={openSearchModal}
+              dropdown={dropdown}
+            />
+            {isAddedToExistingCase ? (
```
[the remark this answers is not in the document]
Comment (stephenhand) [reply plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx:176]: You will also need to add a check on the `enable_case_merging` feature flag here once my PR is merged, if you merge this first you can do it in a separate PR

[the remark this answers is not in the document]
Comment (acedeywin) [reply plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx]: @stephenhand  - Feedback has been implemented.

File: plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx [hunk 3]
```diff
@@ -140,33 +168,70 @@ const BottomBar: React.FC<
...
+                      {!dropdown && (
+                        <KeyboardArrowDownIcon
+                          style={{ fontSize: '20px', marginLeft: '10px', width: '24px', height: '24px' }}
+                        />
+                      )}
+                    </StyledNextStepButton>
+                  </Box>
+                )}
+            {isAddedToCase ? (
+              <SaveAndEndContactButton
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx:213]: I think from the mocks the save button needs to be red whether the contact has been added to a case or not, so you don't need 2 different buttons depending on the state

Remember that the red button also needs the ` {isSubmitting ? <CircularProgress size={12} style={{ position: 'absolute' }} /> : null}` spinner code from the previous save button including

Comment (acedeywin) [reply plugin-hrm-form/src/components/tabbedForms/BottomBar.tsx]: @stephenhand  - It's done.

**Review by stephenhand — APPROVED** [review]

**Pull request merged**